Src/FluentAssertions/Common/Services.cs

```diff
@@ -33,14 +39,76 @@ public static Configuration Configuration
 
     public static IReflector Reflector { get; set; }
 
+    [PublicAPI]
     public static void ResetToDefaults()
     {
-        Reflector = new FullFrameworkReflector();
+        isInitialized = false;
+        EnsureInitialized();
+    }
+
+    internal static void EnsureInitialized()
+    {
+        if (isInitialized)
+        {
+            return;
+        }
+
+        lock (Lockable)
+        {
+            if (!isInitialized)
+            {
+                ExecuteCustomInitializers();
+
+                Reflector = new FullFrameworkReflector();
 #if NETFRAMEWORK || NET6_0_OR_GREATER
-        ConfigurationStore = new ConfigurationStoreExceptionInterceptor(new AppSettingsConfigurationStore());
+                ConfigurationStore = new ConfigurationStoreExceptionInterceptor(new AppSettingsConfigurationStore());
 #else
-        ConfigurationStore = new NullConfigurationStore();
+                    ConfigurationStore = new NullConfigurationStore();
 #endif
-        ThrowException = new TestFrameworkProvider(Configuration).Throw;
+                ThrowException = new TestFrameworkProvider(Configuration).Throw;
+
+                isInitialized = true;
+            }
+        }
+    }
+
+    private static void ExecuteCustomInitializers()
+    {
+        var currentAssembly = Assembly.GetExecutingAssembly();
+        var currentAssemblyName = currentAssembly.GetName();
+
+        var attributes = Array.Empty<AssertionEngineInitializerAttribute>();
+
+        try
+        {
+            attributes = AppDomain.CurrentDomain
+                .GetAssemblies()
+                .Where(assembly => assembly != currentAssembly && !assembly.IsDynamic && !IsFramework(assembly))
+                .Where(a => a.GetReferencedAssemblies().Any(r => r.FullName == currentAssemblyName.FullName))
+                .SelectMany(a => a.GetCustomAttributes<AssertionEngineInitializerAttribute>())
+                .ToArray();
+        }
+        catch
+        {
+            // Just ignore any exceptions that might happen while trying to find the attributes
+        }
+
+        foreach (var attribute in attributes)
+        {
+            try
+            {
+                attribute.Initialize();
+            }
+            catch
+            {
+                // Just ignore any exceptions that might happen while trying to find the attributes
+            }
+        }
+    }
+
+    private static bool IsFramework(Assembly assembly)
+    {
+        return assembly?.FullName?.StartsWith("Microsoft", StringComparison.OrdinalIgnoreCase) == true ||
+            assembly?.FullName?.StartsWith("System", StringComparison.OrdinalIgnoreCase) == true;
```

Comment:
I'm not sure how Qodana concludes that `FullName` cannot be null, but on the other hand I cannot find a realistic case where it can be null.

I guess one would have to really force it to be null.
```cs
class MyAssembly : Assembly
{
    public override string? FullName => null;
}
```

https://stackoverflow.com/questions/70569074/under-what-conditions-can-assembly-getname-name-be-null
suggests that having an assembly without a name is violating the ECMA spec.

Grepping the BCL for `'assembly.*\.FullName\!` gives 17 results.

I'm fine with either of keeping or removing the null checks.